pkg/lib/authn/identity/passkey/provider.go

```diff
@@ -35,8 +34,8 @@ func (p *Provider) Get(userID, id string) (*identity.Passkey, error) {
 	return p.Store.Get(userID, id)
 }
 
-func (p *Provider) GetByAssertionResponse(assertionResponse []byte) (*identity.Passkey, error) {
-	credentialID, err := p.Passkey.GetCredentialIDFromAssertionResponse(assertionResponse)
+func (p *Provider) GetBySpec(spec *identity.PasskeySpec) (*identity.Passkey, error) {
+	_, credentialID, _, err := p.Passkey.PeekAttestationResponse(spec.AttestationResponse)
```

Comment:
No, this is incorrect. Before the change, GetByAssertionResponse is called when the caller has a assertion response. If we rename it to GetBySpec, then it is expected that the caller can call GetBySpec with either a an attestation response, or a assertion response. We need to handle both case.